python/shotgun_desktop/startup.py

```diff
@@ -450,6 +459,8 @@ def pre_engine_start_callback(ctx):
 
         splash.set_message("Launching Engine...")
 
+        __restore_global_debug_flag()
```

Comment:
github doesn't expand enough. This is done during the pre_engine_start_callback of the ToolkitManager.